brayns/Brayns.cpp

```diff
@@ -865,45 +864,46 @@ struct Brayns::Impl : public PluginAPI
 
     void _diffuseShading()
     {
-        RenderingParameters& renderParams =
-            _parametersManager.getRenderingParameters();
-        renderParams.setShading(ShadingType::diffuse);
+        _engine->getRenderer().updateProperty("shadingEnabled", true);
+        _engine->getRenderer().updateProperty("electronShading", false);
     }
 
     void _electronShading()
     {
-        RenderingParameters& renderParams =
-            _parametersManager.getRenderingParameters();
-        renderParams.setShading(ShadingType::electron);
+        _engine->getRenderer().updateProperty("shadingEnabled", false);
+        _engine->getRenderer().updateProperty("electronShading", true);
     }
 
     void _disableShading()
     {
-        RenderingParameters& renderParams =
-            _parametersManager.getRenderingParameters();
-        renderParams.setShading(ShadingType::none);
+        _engine->getRenderer().updateProperty("shadingEnabled", false);
+        _engine->getRenderer().updateProperty("electronShading", false);
     }
 
     void _increaseAmbientOcclusionStrength()
     {
-        RenderingParameters& renderParams =
-            _parametersManager.getRenderingParameters();
-        float aaStrength = renderParams.getAmbientOcclusionStrength();
-        aaStrength += 0.1f;
-        if (aaStrength > 1.f)
-            aaStrength = 1.f;
-        renderParams.setAmbientOcclusionStrength(aaStrength);
+        auto& renderer = _engine->getRenderer();
+        if (!renderer.hasProperty("aoWeight"))
+            return;
+
+        auto aoStrength = renderer.getProperty<float>("aoWeight");
+        aoStrength += 0.1f;
+        if (aoStrength > 1.f)
+            aoStrength = 1.f;
+        renderer.updateProperty("aoWeight", aoStrength);
     }
 
     void _decreaseAmbientOcclusionStrength()
     {
-        RenderingParameters& renderParams =
-            _parametersManager.getRenderingParameters();
-        float aaStrength = renderParams.getAmbientOcclusionStrength();
-        aaStrength -= 0.1f;
-        if (aaStrength < 0.f)
-            aaStrength = 0.f;
-        renderParams.setAmbientOcclusionStrength(aaStrength);
+        auto& renderer = _engine->getRenderer();
+        if (!renderer.hasProperty("aoWeight"))
+            return;
+
+        auto aoStrength = renderer.getProperty<float>("aoWeight");
+        aoStrength -= 0.1f;
+        if (aoStrength < 0.f)
+            aoStrength = 0.f;
+        renderer.updateProperty("aoWeight", aoStrength);
```

Comment:
Strictly speaking, those actions should not be in the core, hence it becomes client code where those strings are either hardcoded or better, are taken from the schema. This is the case for the web clients. This change just tries to avoid breaking existing behavior. I can define a string constant here, but it's still wrong :)